Yoast/wordpress-seo
# Show excerpt snippet preview

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the excerpt replacement variable will output a piece of the post content when no excerpt is given.

## Test instructions

This PR can be tested by following these steps:

* On trunk set `%%excerpt%%` as a template for the metadescription or have it as a value in the metadescription field for a post itself (this fix doesn't affect terms).
* When no excerpt is given nothing will be replaced. But if you save the post, go to the frontend, you'll see that `<meta name="description"` is filled with the first part of your post. 
* Checkout this branch.
* Keep `%%excerpt%%` as a template for the metadescription or as a value in the metadescription field for a post itself.
* Have some post content and empty the excerpt field. Now you will see a partial from the post content being shown.
* Save the post and go to the frontend.  You'll see that `<meta name="description"` is filled with the first part of your post. 
* When you use the `excerpt only` snippet variable, you shouln't see a part of the text.

NB. test in Gutenberg and the classic editor.

## Quality assurance


Fixes #5135
